### ISSUE TITLE
YoloV8 Instance Segmentation training doesn't learn on Windows 11 Conda Environment

### ISSUE THREAD

**Issue by AhmetEkiz** [ultralytics/ultralytics#3123]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

The CLI command for training:
`yolo task=segment mode=train model=yolov8n-seg.pt data=C:\\Users\\ekizo\\projects_2\\yolov8\\tomato-train-100-5\data.yaml epochs=10 imgsz=640`

This is the output of the Yolov8 training on Windows 11 Conda Environment.

```
(torch) C:\Users\ekizo\projects_2\yolov8>yolo task=segment mode=train model=yolov8n-seg.pt data=C:\\Users\\ekizo\\projects_2\\yolov8\\tomato-train-100-5\\data.yaml epochs=10 imgsz=640
Ultralytics YOLOv8.0.28  Python-3.9.16 torch-2.0.1 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
yolo\engine\trainer: task=segment, mode=train, model=yolov8n-seg.pt, data=C:\\Users\\ekizo\\projects_2\\yolov8\\tomato-train-100-5\\data.yaml, epochs=10, patience=50, batch=16, imgsz=640, save=True, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=False, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, image_weights=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, hide_labels=False, hide_conf=False, vid_stride=1, line_thickness=3, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, fl_gamma=0.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=False, save_dir=runs\segment\train23
Overriding model.yaml nc=80 with nc=10

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.Conv                  [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.Conv                  [16, 32, 3, 2]
  2                  -1  1      7360  ultralytics.nn.modules.C2f                   [32, 32, 1, True]
  3                  -1  1     18560  ultralytics.nn.modules.Conv                  [32, 64, 3, 2]
  4                  -1  2     49664  ultralytics.nn.modules.C2f                   [64, 64, 2, True]
  5                  -1  1     73984  ultralytics.nn.modules.Conv                  [64, 128, 3, 2]
  6                  -1  2    197632  ultralytics.nn.modules.C2f                   [128, 128, 2, True]
  7                  -1  1    295424  ultralytics.nn.modules.Conv                  [128, 256, 3, 2]
  8                  -1  1    460288  ultralytics.nn.modules.C2f                   [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.SPPF                  [256, 256, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.Concat                [1]
 12                  -1  1    148224  ultralytics.nn.modules.C2f                   [384, 128, 1]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.Concat                [1]
 15                  -1  1     37248  ultralytics.nn.modules.C2f                   [192, 64, 1]
 16                  -1  1     36992  ultralytics.nn.modules.Conv                  [64, 64, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.Concat                [1]
 18                  -1  1    123648  ultralytics.nn.modules.C2f                   [192, 128, 1]
 19                  -1  1    147712  ultralytics.nn.modules.Conv                  [128, 128, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.Concat                [1]
 21                  -1  1    493056  ultralytics.nn.modules.C2f                   [384, 256, 1]
 22        [15, 18, 21]  1   1006030  ultralytics.nn.modules.Segment               [10, 32, 64, [64, 128, 256]]
YOLOv8n-seg summary: 261 layers, 3265566 parameters, 3265550 gradients, 12.1 GFLOPs

Transferred 381/417 items from pretrained weights
optimizer: SGD(lr=0.01) with parameter groups 66 weight(decay=0.0), 77 weight(decay=0.0005), 76 bias
train: Scanning C:\Users\ekizo\projects_2\yolov8\tomato-train-100-5\train\labels... 800 images, 9 backgrounds, 0 corrupt: 100%|█████
train: New cache created: C:\Users\ekizo\projects_2\yolov8\tomato-train-100-5\train\labels.cache
val: Scanning C:\Users\ekizo\projects_2\yolov8\tomato-train-100-5\valid\labels... 200 images, 1 backgrounds, 0 corrupt: 100%|███████
val: New cache created: C:\Users\ekizo\projects_2\yolov8\tomato-train-100-5\valid\labels.cache
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to runs\segment\train23
Starting training for 10 epochs...
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       1/10      2.75G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [02:31<00:00,  3.02
C:\Users\ekizo\anaconda3\envs\torch\lib\site-packages\torch\optim\lr_scheduler.py:139: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95)
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       2/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [02:24<00:00,  2.90
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95)
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       3/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [02:27<00:00,  2.94
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95)
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       4/10       3.4G        nan        nan        nan        nan         15        640: 100%|██████████| 50/50 [02:29<00:00,  3.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       5/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [02:26<00:00,  2.93s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       6/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [02:33<00:00,  3.06s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       7/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [02:10<00:00,  2.60s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       8/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [01:27<00:00,  1.75s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
       9/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [01:23<00:00,  1.67s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
      10/10       3.4G        nan        nan        nan        nan         16        640: 100%|██████████| 50/50 [01:23<00:00,  1.67s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0

10 epochs completed in 0.715 hours.
Optimizer stripped from runs\segment\train23\weights\last.pt, 6.8MB
Optimizer stripped from runs\segment\train23\weights\best.pt, 6.8MB

Validating runs\segment\train23\weights\best.pt...
Ultralytics YOLOv8.0.28  Python-3.9.16 torch-2.0.1 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
YOLOv8n-seg summary (fused): 195 layers, 3260014 parameters, 0 gradients, 12.0 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████
                   all        200        200          0          0          0          0          0          0          0          0
Speed: 3.1ms pre-process, 50.6ms inference, 0.0ms loss, 0.2ms post-process per image
Results saved to runs\segment\train23

(torch) C:\Users\ekizo\projects_2\yolov8>
```
- **Ultralytics settings file from C:\Users\ekizo\AppData\Roaming\Ultralytics. (It's actually yaml.):** [settings.txt](https://github.com/ultralytics/ultralytics/files/11711745/settings.txt)
- **The dataset data.yaml file:** [data.txt](https://github.com/ultralytics/ultralytics/files/11711749/data.txt)
- **Conda Environment File:** [environment.txt](https://github.com/ultralytics/ultralytics/files/11711774/environment.txt)

| val_batch1_labels |val_batch1_pred |
| ----------------- | ----------------- |
|![val_batch1_labels](https://github.com/ultralytics/ultralytics/assets/33065334/061ef53f-8b7c-4e13-be92-469a1a423900)   |  ![val_batch1_pred](https://github.com/ultralytics/ultralytics/assets/33065334/437c8755-d7e4-43c4-9d1b-59d5a2b56c11) |
       

### Environment

Ultralytics YOLOv8.0.28  Python-3.9.16 torch-2.0.1 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
- OS: Windows 11
- Conda Environment
- YOLO Ultralytics: 8.0.28
- Pytorch: 2.0.1
- Python 3.9.16

### Minimal Reproducible Example

!pip install ultralytics==8.0.28

rf = Roboflow(api_key="YOUR API")
project = rf.workspace("samkeun5").project("tomato-train-100")
dataset = project.version(5).download("yolov8")

!yolo task=segment mode=train model=yolov8n-seg.pt data=C:\\Users\\ekizo\\projects_2\\yolov8\\tomato-train-100-5\data.yaml epochs=10 imgsz=640

### Additional

The Yolov8 Instance Segmentation model doesn't learn anything on Windows 11 Conda Environment but it learns on Colab. 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@AhmetEkiz 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – **Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.**
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by glenn-jocher** [comment]
@AhmetEkiz most relevant section above is:

 **✅ Current – Verify that your code is up-to-date with GitHub with git pull or pip install -U ultralytics to ensure your problem has not already been solved in the latest code version.**

**Comment by glenn-jocher** [comment]
Great news 😃! Your original issue may now be resolved ✅ in PR #3197. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by AhmetEkiz** [comment]
Hi @glenn-jocher again, thanks for your help. 😃 
I updated with `pip install -U ultralytics` but it didn't work.

I installed it from repo but it didn't work too.
```
git clone https://github.com/ultralytics/ultralytics
cd ultralytics
pip install -e .
```

I downgraded the torch to 1.7.0 but it didn't work too. 

I solved some errors while applying these steps but it didn't work anyway. Maybe I need to use Docker Container or Linux. But I am surprised that it didn't work.

**Comment by fuweifu-vtoo** [comment]
Dear author, this problem seems to have not been solved yet. It has been several months. Is there any latest progress?

**Comment by glenn-jocher** [comment]
@fuweifu-vtoo hi, thank you for reaching out again and I apologize for the delay in resolving this issue. Our team is actively working on it, but unfortunately, there's no specific timeframe for a fix yet. We understand the inconvenience this might be causing and highly appreciate your patience and continued support. We'll make sure to post updates regarding the progress in this thread. Thanks!
